packages/uikit/src/hooks/useTooltip/useTooltip.tsx

```diff
@@ -206,30 +198,38 @@ const useTooltip = (content: React.ReactNode, options?: TooltipOptions): Tooltip
     ],
   });
 
-  const tooltip = (
-    <StyledTooltip
-      data-theme={isDark ? "light" : "dark"}
-      {...animationMap}
-      variants={animationVariants}
-      transition={{ duration: 0.3 }}
-      ref={setTooltipElement}
-      style={styles.popper}
-      {...attributes.popper}
-    >
-      {content}
-      <Arrow ref={setArrowElement} style={styles.arrow} />
-    </StyledTooltip>
+  const tooltip = useMemo(() => {
+    return (
+      <StyledTooltip
+        data-theme={isDark ? "light" : "dark"}
+        {...animationMap}
+        variants={animationVariants}
+        transition={{ duration: 0.3 }}
+        ref={setTooltipElement}
+        style={styles.popper}
+        {...attributes.popper}
+      >
+        {content}
+        <Arrow ref={setArrowElement} style={styles.arrow} />
+      </StyledTooltip>
+    );
+  }, [attributes?.popper, content, isDark, styles?.arrow, styles?.popper]);
```

Comment:
If we want to prevent it from rerendering if no prop changes, shouldn't we apply `React.memo` to the component?

Reply:
> such as transition prop in StyledTooltip which gets a new object everytime that makes react.memo to recompute element on each render

I think we need to prevent passing new object reference to the component on each render in the first place? IMO if we're writing custom checker to bypass, maybe we should reconsider the way we design the props of that component.  Because fundamentally the way how react works is to reflect the props changes to the UI, which means rerender the component whenever the props updated right?

Reply:
another benefit for `memo` over `useMemo` imo is we don't have to write `useMemo` every time we use that particular component